stripe/src/main/java/com/stripe/android/model/SourceParams.java

```diff
@@ -375,19 +388,114 @@ public String getType() {
     }
 
     /**
+     * @return a custom type of this source, if one has been set
+     */
+    @Nullable
+    public String getCustomType() {
+        return mCustomType;
+    }
+
+    /**
      * @return the custom metadata set on these params
      */
     public Map<String, String> getMetaData() {
         return mMetaData;
     }
 
+    /*---- Setters ----*/
+
+    /**
+     * @param amount currency amount for this source, in the lowest denomination.
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setAmount(long amount) {
+        mAmount = amount;
+        return this;
+    }
+
+    /**
+     * @param apiParameterMap a map of parameters specific for this type of source
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setApiParameterMap(
+            @NonNull Map<String, Object> apiParameterMap) {
+        mApiParameterMap = apiParameterMap;
+        return this;
+    }
+
+    /**
+     * @param currency currency code for this source (i.e. "EUR")
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setCurrency(String currency) {
+        mCurrency = currency;
+        return this;
+    }
+
+    /**
+     * @param owner an {@link SourceOwner} object for this source
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setOwner(Map<String, Object> owner) {
+        mOwner = owner;
+        return this;
+    }
+
+    /**
+     * Sets a redirect property map for this source object. If you only want to
+     * set a return url, use {@link #setReturnUrl(String)}.
+     *
+     * @param redirect a set of redirect parameters
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setRedirect(Map<String, Object> redirect) {
+        mRedirect = redirect;
+        return this;
+    }
+
+
+    /**
+     * @param returnUrl a redirect URL for this source.
+     * @return {@code this}, for chaining purposes
+     */
+    public SourceParams setReturnUrl(@NonNull @Size(min = 1) String returnUrl) {
+        setRedirect(createSimpleMap(FIELD_RETURN_URL, returnUrl));
```

Comment:
Ok. The main reason for this method is that the only reason it's a map at all is because the JSON model has redirect as an object, but one that only ever has one key. I translate JSON objects to Map<String, Object>. Theoretically there can be other fields though. Will fix.